### PR TITLE
contrib/kafka: take env variable into account to enable DSM

### PR DESCRIPTION
**Pull request by vandonr**
In most tracers, setting the `DD_DATA_STREAMS_ENABLED` env variable is enough to enable DSM, but in go this has to be done at wrapping time, which is inconsistent with other languages.

In this change, I deprecated the option to enable DSM, in favor of the env variable.

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by piochelepiotr — APPROVED** [review]

File: contrib/Shopify/sarama/option.go [hunk 1]
```diff
@@ -54,6 +56,8 @@ func WithServiceName(name string) Option {
 }
 
 // WithDataStreams enables the Data Streams monitoring product features: https://www.datadoghq.com/product/data-streams-monitoring/
+//
+// Deprecated: Use the environment variable DD_DATA_STREAMS_ENABLED instead
```
Comment (rarguelloF) [inline contrib/Shopify/sarama/option.go:60]: why can't we keep both? I think its a common pattern in the Go tracer to provide both in-code and env variable choices for configurations

**Review by rarguelloF — COMMENTED** [review]
LGTM! just a small note about the deprecated options

File: contrib/Shopify/sarama/option.go (hunk 1 above)
Comment (vandonr) [reply contrib/Shopify/sarama/option.go:60]: hmm, yes that's an argument that can be heard. Though it lets go users do stuff that won't work in other tracers, which is not _the best_, but being consistent within the go tracer itself is probably more important.

**Review by dianashevchenko — APPROVED** [review]

**Pull request merged**